src/main/groovy/com/vanniktech/maven/publish/MavenPublishPlugin.groovy

```diff
@@ -39,10 +39,28 @@ class MavenPublishPlugin extends BaseMavenPublishPlugin {
       options.linksOffline "https://developer.android.com/reference", "${project.android.sdkDirectory}/docs/reference"
     }
 
-    project.tasks.create("androidJavadocsJar", Jar.class) {
+    def androidJavadocsJar = project.tasks.create("androidJavadocsJar", Jar.class) {
       classifier = 'javadoc'
-      from project.androidJavadocs.destinationDir
-    }.dependsOn("androidJavadocs")
+    }
+
+    if (plugins.hasPlugin('kotlin-android')) {
+      def dokkaOutput = "${project.docsDir}/dokka"
+      project.plugins.apply('org.jetbrains.dokka-android')
+      project.dokka {
+        outputFormat 'javadoc'
+        outputDirectory dokkaOutput
+      }
+      androidJavadocsJar.dependsOn("dokka")
+      androidJavadocsJar.configure {
+        dependsOn "dokka"
```

Comment:
Does this duplicate the code on line 53? I think `androidJavadocsJar.dependsOn("dokka")` is equivalent to `androidJavadocsJar.configure { dependsOn "dokka" }`.